experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -85,64 +85,6 @@ struct HandshakeSpeculationPass
 };
 } // namespace
 
-template <typename T>
-LogicalResult HandshakeSpeculationPass::placeUnits(Value ctrlSignal) {
-  MLIRContext *ctx = &getContext();
-  OpBuilder builder(ctx);
-
-  for (OpOperand *operand : placements.getPlacements<T>()) {
-    Operation *dstOp = operand->getOwner();
-    Value srcOpResult = operand->get();
-
-    // Create and connect the new Operation
-    builder.setInsertionPoint(dstOp);
-    // resultType is tentative and will be updated in the addSpecTag algorithm
-    // later.
-    T newOp =
-        builder.create<T>(dstOp->getLoc(), /*resultType=*/srcOpResult.getType(),
-                          /*dataIn=*/srcOpResult, /*ctrl=*/ctrlSignal);
-    inheritBB(dstOp, newOp);
-
-    // Connect the new Operation to dstOp
-    // Note: srcOpResult.replaceAllUsesExcept cannot be used here
-    // The following bug may occur in most cases:
-    // (a) Consider a scenario where a control value from a buffer is passed to
-    // the control branch.
-    // (b) Simultaneously, a speculator uses the same control value from the
-    // buffer as a trigger signal.
-    // (c) A save-commit unit is positioned on the edge from the buffer to the
-    // control branch.
-    // (d) If we apply replaceAllUsesExcept to the value referenced by the
-    // save-commit unit, the speculator will also be placed after the
-    // save-commit unit, which is undesirable.
-    operand->set(newOp.getResult());
```

Comment:
I removed this long comment in the new methods, because `operand->set` is a natural choice, rather than `replaceAllUsesExcept`.